tests/mir-opt/instcombine_duplicate_switch_targets_e2e.rs

```diff
@@ -0,0 +1,16 @@
+// compile-flags: -Zmir-opt-level=2 -Zinline-mir
+// ignore-debug: standard library debug assertions add a panic that breaks this optimization
```

Comment:
Blessing the `ignore-debug` tests requires to recompile everything, which is tedious.
Can you use the intrinsic directly and cook your own `unreachable_unchecked`?

Reply:
I'm really wary of doing things like that. Modifying the mir-opt test for the convenience of testing is what caused this regression in the first place, and this test is already somewhat removed from the code that was reported to not optimize.